### PR TITLE
Make code organization match the functional structure

### PR DESCRIPTION
**Pull request by alamb**
part of https://github.com/datafusion-contrib/datafusion-tui/issues/132

We are moving `dft` so there is an engine and 2 uis (the terminal tui and the cli)

Let's make the code structure match this functional structure:
1. Rename `ui` to `tui` to make it clear it is the terminal ui
2. Move the CLI related code into its own module (cli)
3. Move the command line parsing structures into `DftArgs` in in main.rs

I don't fully understand the split between what logic belongs in `app` and what belongs in `tui` but I figure we can work on that split over time

### CODE REVIEW

File: src/app/handlers/mod.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ use tonic::transport::Channel;
 
 use crate::{
     app::{state::tabs::history::HistoryQuery, AppEvent},
-    ui::SelectedTab,
+    tui::SelectedTab,
```
Comment (alamb) [inline src/app/handlers/mod.rs:38]: I renamed `ui` to `tui`

File: src/app/mod.rs [hunk 2]
```diff
@@ -88,15 +80,14 @@ pub struct App<'app> {
 }
 
 impl<'app> App<'app> {
-    pub fn new(state: state::AppState<'app>, cli: DftCli) -> Self {
+    pub fn new(state: state::AppState<'app>) -> Self {
```
Comment (alamb) [inline src/app/mod.rs:83]: The cli arguments don't appear to be used after construction so I removed them from the structures

File: src/app/mod.rs [hunk 3]
```diff
@@ -329,118 +320,3 @@ pub async fn run_app(cli: cli::DftCli, state: state::AppState<'_>) -> Result<()>
     }
     app.exit()
 }
-
-/// Encapsulates the command line interface
-pub struct CliApp {
```
Comment (alamb) [inline src/app/mod.rs:334]: moved into `cli`

File: src/cli/mod.rs [hunk 4]
```diff
@@ -14,82 +14,132 @@
...
+use color_eyre::eyre::eyre;
+use datafusion::arrow::util::pretty::pretty_format_batches;
+use datafusion::execution::SendableRecordBatchStream;
+use datafusion::sql::parser::DFParser;
+use futures::StreamExt;
+use log::info;
+use crate::app::state;
+use crate::execution::ExecutionContext;
 
-use clap::Parser;
```
Comment (alamb) [inline src/cli/mod.rs:20]: I moved the CLI parsing into main.rs

File: src/app/handlers/mod.rs (hunk 1 above)
Comment (matthewmturner) [reply src/app/handlers/mod.rs:38]: i wont have time for a full review right now, but as a drive by comment i think what would align more to the structure i have in mind is renaming `app` to `tui`.  The `ui` code is actual display code for the `tui` app (i.e. it is `ratatui` widgets like `Table`, `Textarea`, `Paragraph`, etc.). i can imagine having a directory structure something like this:

tui/state
tui/handlers
tui/ui
cli/...
execution/...
...

File: src/main.rs [hunk 5]
```diff
@@ -38,9 +40,78 @@ async fn main() -> Result<()> {
...
-        let state = state::initialize(cli.clone());
-        run_app(cli.clone(), state).await?;
+        let state = state::initialize(cli.config_path());
+        run_app(state).await?;
     }
 
     Ok(())
 }
+
+const LONG_ABOUT: &str = "
```
Comment (matthewmturner) [inline src/main.rs:50]: doesnt need to be in this PR but maybe we put all of the clap logic in something like `args.rs`?

File: src/app/handlers/mod.rs (hunk 1 above)
Comment (alamb) [reply src/app/handlers/mod.rs]: good call -- done in 9ea59d6

File: src/main.rs (hunk 5 above)
Comment (alamb) [reply src/main.rs]: Done in 5e88d0a

**Comment by alamb** [comment]
I had to get the PR checked out to fix the formatting, so I went ahead and made the changes you suggested 

While I was at it, I also noticed that the `config` is not really part of the terminal ui so I moved it from `app/config.rs` to `config.rs`

**Pull request merged**